konflux-ci/build-definitions
# Attach migration file to task bundle

[STONEBLD-3026](https://issues.redhat.com//browse/STONEBLD-3026)


* Introduce concepts of normal task (e.g. buildah) and extended task
  (e.g. buildah-20gb, buildah-min).
* Refactor the code that processes task build and push.
* For testing purpose, add a simple local cache in order to reduce the
  number of skopeo-inpsect runs for checking existence of task bundles.
  This could also be helpful to reduce the whole script execution during
  testing.
* For testing purpose, script can be configured with environment
  variable TEST_TASKS to run and check the result for a small number of
  tasks.

To test the attach:

* Open a new branch.
* Bump version of a task in the YAML definition, e.g. task/buildah/0.2
* Create task/buildah/0.2/migrations/<the new version>.sh
* Commit the changes
* Run hack/build-and-push.sh by configuring it to push images to your
  own test image repository.
* Find out the buildah task bundle.
* Run `oras discover \<bundle reference>'. The attached migration file
  should be listed.
